### PR TITLE
Move AnsibleLintRule inside rules module

### PR DESCRIPTION
**Pull request by ssbarnea**
Moved code to ease reusability and avoid cyclic imports.

### CODE REVIEW

File: lib/ansiblelint/__init__.py [hunk 1]
```diff
@@ -22,123 +22,15 @@
 from collections import defaultdict
 import logging
 import os
-import re
 
+from ansiblelint.rules import AnsibleLintRule  # noqa
```
Comment (webknjaz) [inline lib/ansiblelint/__init__.py:26]: Please avoid `noqa` unless there's a serious reason to do so.

Comment (ssbarnea) [reply lib/ansiblelint/__init__.py:26]: That is keeping the API compatibility with older versions. Any custom rules would import the base class from here, as that was also included inside the docs.

AnsibleLintRule is likely the only effectively public API for the linter, others stuff are likely to be considered private.

File: lib/ansiblelint/__init__.py [hunk 2]
```diff
@@ -22,123 +22,15 @@
 from collections import defaultdict
 import logging
 import os
-import re
 
+from ansiblelint.rules import AnsibleLintRule # noqa F401 backwards compatibility
```
Comment (webknjaz) [inline lib/ansiblelint/__init__.py:26]: I'm pretty sure that the correct syntax would be
```suggestion
from ansiblelint.rules import AnsibleLintRule  # noqa: F401 
 # exposing the public API
```

File: lib/ansiblelint/__init__.py (hunk 1 above)
Comment (ssbarnea) [reply lib/ansiblelint/__init__.py]: I improved the exception, explaining it.

File: lib/ansiblelint/generate_docs.py [hunk 3]
```diff
@@ -60,15 +58,10 @@ def import_all_rules():
...
-    class_list = []
-    for mod in mod_list:
-        class_temp = [m[1] for m in getmembers(mod) if isclass(m[1])]
-        class_temp = [c for c in class_temp if c is not AnsibleLintRule]
-        class_list.extend(class_temp)
+    # called after all classes are loaded
 
     all_rules = []
-    for c in class_list:
+    for c in AnsibleLintRule.__subclasses__():
```
Comment (webknjaz) [inline lib/ansiblelint/generate_docs.py:64]: I'm not sure if this'd work because you'd have to import all the modules in this runtime first.

Comment (webknjaz) [reply lib/ansiblelint/generate_docs.py:64]: So I'd leave this change out of the PR.

Comment (ssbarnea) [reply lib/ansiblelint/generate_docs.py:64]: This is `what import_all_rules` does just before calling `get_serialized_rules`. The proof that it works is that the docs are not affected by this change.

I do have many questions regarding how the  generate_docs.py was originally implemented but that would be subject of other changes. For the moment I did only what was needed in order to allow me to move AnsibleLintRule definition inside.

**Pull request merged**